### ISSUE TITLE
Add Tooltip to SidePanel (labs) Control Button

### ISSUE THREAD

**Issue by lychyi** [Workday/canvas-kit#909]
## 🚀 Feature Proposal

The Workday a11y team has issued guidelines that further helps this button's accessibility: **add a tooltip to the button to increase its affordance as the icon visual alone may not be sufficient for sighted users.**

![image](https://user-images.githubusercontent.com/14299381/99298401-8450db00-27fe-11eb-8106-a1a30f813f52.png)

## Motivation
Our Labs side panel ships with a control button that is an `IconButton` that uses `aria-labelledby` or `aria-label` to convey what it is meant to do. This is made easier through our `useSidePanel` hook which assumes there's an element that contains the label for the button. Today's use of SidePanel looks like [this](https://github.com/Workday/canvas-kit/tree/master/modules/_labs/side-panel/react#usage):

```jsx
import SidePanel, {useSidePanel} from '@workday/canvas-kit-labs-react-side-panel';

/**
 * A SidePanel is made up of three components
 * - The panel itself <SidePanel>
 * - An accessible name (either an existing element or you can use <span hidden>)
 * - A control that expands/collapses the SidePanel, usually <SidePanel.ToggleButton> as a child of <SidePanel>
 * A convenience hook is available for setting the state of the SidePanel along with the proper aria- attributes
 *
 */
const {panelProps, labelProps, controlProps} = useSidePanel();

<SidePanel {...panelProps}>
  <SidePanel.ToggleButton {...controlProps} />
  <h3 {...labelProps}>Tasks Panel</h3>
  {/* Your panel contents */}
</SidePanel>
```

Our current `Tooltip` component [alters the element it wraps](https://github.com/Workday/canvas-kit/tree/master/modules/tooltip/react#optional) (in this case, it'll be the `SidePanel.ToggleButton`) with an `aria-label` or `aria-describedby`. Generally speaking, [`aria-labelledby` takes precedence over `aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute) so doing something like this:

```jsx
...
<Tooltip title={expanded ? `Collapse` : `Expand`}>
  <SidePanel.ToggleButton {...controlProps} />
</Tooltip>
...
```

...would still be accessible because our hook spreads `aria-labelledby` by default. But there can be two problems:
1. the underlying button will have both an `aria-labelledby` and `aria-label` (and they may be different values too)
2. (if a user doesn't use our SidePanel hook) the underlying button would have its `aria-label` overridden by `Tooltip`

This makes using our current `Tooltip` component for this use case a bit messy. It's do-able but there are a few gotchas to consider.

## Proposal
One way around this is to create a new `SidePanel` member called `ToggleButtonWithTooltip` (or something more cleverly named) that ships a custom `Tooltip` wrapper that does not automatically add an `aria-` prop to the wrapped button.

or

We can augment `Tooltip` or ship another `Tooltip` that does not automatically wrap its elements with an `aria-` attribute.

## Example

```jsx
<SidePanel {...panelProps}>
  <SidePanel.ToggleButtonWithTooltip title={expanded ? `Collapse` : `Expand`} {...controlProps} />
  <h3 {...labelProps}>Tasks Panel</h3>
  {/* Your panel contents */}
</SidePanel>
```

**Comment by NicholasBoll** [comment]
#980 added the `muted` type which doesn't add any `aria-*` attributes (meaning the `aria-*` will be handled by the developer). I don't think Tooltip has to change to accommodate this anymore.

**Comment by NicholasBoll** [comment]
Closed by #1401
